### ISSUE TITLE
Menu Rule provider

### ISSUE THREAD

**Issue by ns8482e** [OrchardCMS/OrchardCore#17193]
Allow Site administrator to manage Admin menu according to their needs.

- Provide a Menu Rule provider json file  `menu.json` similar to the placement
- Allow author to write rules to show/hide/position of the placement in the menu.json
- Navigation provider - evaluate the menu rule before adding menu to the menu items.

This will allow site administrator to control menu items and their order without change in orchard core code base.

Ref
https://github.com/OrchardCMS/OrchardCore/pull/17112#issuecomment-2529057880

**Comment by MikeAlhayek** [comment]
I agree with this. It would be great to be able to use existing placements structure to define the position of each menu level/menu-item.

**Comment by sebastienros** [comment]
Configuration then. Deployed with the app. Like placement. And optional, meaning it shouldn't make menus implementation harder, only for the people who don't like @MikeAlhayek 's menu.

**Comment by ns8482e** [comment]
> Configuration then. Deployed with the app. Like placement. And optional, meaning it shouldn't make menus implementation harder, only for the people who don't like @MikeAlhayek 's menu.

New structure is fine as long as it's applies to new sites, I personally prefer to have upgrade seamless, and new UI should be opt-in
